src/nvim/tui/tui.c

```diff
@@ -1233,6 +1233,9 @@ static void tui_set_mode(TUIData *tui, ModeShape mode)
     HlAttrs aep = kv_A(tui->attrs, c.id);
 
     tui->want_invisible = aep.hl_blend == 100;
+    if (tui->want_invisible || aep.rgb_bg_color == -1) {
+      return;
```

Comment:
I don't think it should just return when the cursor background is invalid, because it still need to set cursor shape in the code below. Maybe the code to set cursor color should be moved into a separate function.